core/src/main/java/org/infinispan/statetransfer/StateConsumerImpl.java

```diff
@@ -301,6 +301,7 @@ public void onTopologyUpdate(final CacheTopology cacheTopology, final boolean is
       }
       stateTransferLock.releaseExclusiveTopologyLock();
       stateTransferLock.notifyTopologyInstalled(cacheTopology.getTopologyId());
+      remoteCommandsExecutor.checkForReadyTasks();
```

Comment:
Unfortunately after the command is processed also means after the cluster listeners and the transaction data were received, which might be too late. But you're right, in my test I didn't have any write operations that could fill the remote executor and cause `checkForReadyTasks()` to block.

I'll try again without this change, and if the stress test fails I'll try to put the `checkForReadyTasks()` call on a different thread.